build/js/live-editor.output_pjs.js

```diff
@@ -1152,6 +1152,157 @@ var BabyHint = {
+    var resourceRecords = this.getResourceRecords(userCode);
+
+    // Insert the images into a hidden div to cause them to load
+    // but not be visible to the user
+    if (!this.imageHolder) {
+        this.imageHolder = $("<div>")
+            .css({
+                height: 0,
+                width: 0,
+                overflow: "hidden",
+                position: "absolute"
+            })
+            .appendTo("body");
+    }
+
+    var promises = resourceRecords.map(this.loadResource.bind(this));
+
+    $.when.apply($, promises).then(callback);
+};
+
+PJSResourceCache.prototype.getResourceRecords = function(userCode) {
+    var resourceRegex = /get(Image|Sound)\s*\(['"](.*?)['"]\)/g;
+
+    var resources = [];
+    var match = resourceRegex.exec(userCode);
+    while (match) {
+        resources.push({
+            filename: match[2],
+            type: match[1].toLowerCase()
+        });
+        match = resourceRegex.exec(userCode);
+    }
+
+    return resources;
+};
+
+PJSResourceCache.prototype.loadResource = function(resourceRecord) {
+    var filename = resourceRecord.filename;
+    switch (resourceRecord.type) {
+        case "image":
+            return this.loadImage(filename);
+            break;
+        case "sound":
+            return this.loadSound(filename);
+            break;
+        default:
+            throw "we can't load '" + type + "' resources yet";
+            break;
+    }
+};
+
+PJSResourceCache.prototype.loadImage = function(filename) {
+    var deferred = $.Deferred();
+    var path = this.output.imagesDir + filename + ".png";
+
+    var img = document.createElement("img");
+    img.onload = function() {
+        this.cache[filename + ".png"] = img;
+        deferred.resolve();
+    }.bind(this);
+    img.onerror = function() {
+        deferred.resolve(); // always resolve
+    }.bind(this);
+
+    img.src = path;
+    this.imageHolder.append(img);
+
+    return deferred;
+};
+
+PJSResourceCache.prototype.loadSound = function(filename) {
+    var deferred = $.Deferred();
+    var audio = document.createElement("audio");
+
+    audio.preload = "auto";
+    audio.oncanplaythrough = function() {
+        this.cache[filename + ".mp3"] = {
+            audio: audio,
+            __id: function () {
+                return "getSound('" + filename + "')";
+            }
+        };
+        deferred.resolve();
+    }.bind(this);
+    audio.onerror = function() {
+        deferred.resolve();
+    }.bind(this);
+
+    audio.src = this.output.soundsDir + filename + ".mp3";
+
+    return deferred;
+};
+
+PJSResourceCache.prototype.getResource = function(filename, type) {
+    switch (type) {
+        case "image":
+            return this.getImage(filename);
+            break;
+        case "sound":
+            return this.getSound(filename);
+            break;
+        default:
+            throw "we can't load '" + type + "' resources yet";
```

Comment:
Ah, okay. We do automatically send JS errors off to Sentry though we dont necessarily use that information yet.